keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,978 @@
++ // A null value indicates that the PodDisruptionBudget is allow to be disrupted by any other pods with a higher priority.
++ // +optiona
++ AllowDisruptionByPriorityGreaterThanOrEqual *int32
+}
+```
+
+The following is an example of a `PriorityClass` where the user sets the `allowDisruptionByPriorityGreaterThanOrEqual` as 1000
+indicating that the pods corresponding to this priorty class can only be violated by the pods with a value of priority 
+greater than or equal to 1000 during the scheduler preemption process
+
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: low-priority
+value: 100
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+allowDisruptionByPriorityGreaterThanOrEqual: 1000
+```
+
+The scheduler plugin `defaultpreemption` needs to check the value set in the `AllowDisruptionByPriorityGreaterThanOrEqual` field 
+when selecting victims.
+- if the priority of the preemptor is greater than or equal to the value of `AllowDisruptionByPriorityGreaterThanOrEqual` in victim pod, 
+  the implementation will remain consistent with the existing behavior, meaning that the scheduler will try to select victims whose PDBs 
+  are not violated by preemption, but if no such victims are found, preemption will still happen and lower priority pods will be preempted 
+  via the `/evictions` endpoint despite their PDBs being violated. If the `/eviction` endpoint returns a response `429 Too Many Requests`
+  and the scheduler will fallback to deletion as an alternative.
+- if the priority of the preemptor is less than the value of `AllowDisruptionByPriorityGreaterThanOrEqual` in victim pod, 
+  the scheduler will check if the victim' PDBs will be violated when selecting victims
+  - if violate the victims' PDBs, this victim will not be selected as candidates.
+  - if not violate the victims' PDBs, scheduler will preempt this pod via the `/evictions` endpoint. If it responds `200 OK`, 
+    it means the eviction is allowed, and the victim is deleted, similar to sending a DELETE request to the Pod URL. If it responses
+    `429 Too Many Requests`, the scheduler will output an error log and choose another victim among the candidate victims to preempt 
+    until it succeeds or there are no more candidates.
+
+### PreemptionPolicy vs AllowDisruptionByPriorityGreaterThanOrEqual
+
+The `PreemptionPolicy` is used to describe the behavior of the pods associated with the PriorityClass during preemption as the preemptor. 
+The `AllowDisruptionByPriorityGreaterThanOrEqual` is used to describe the policy of the pods associated with the PriorityClass during 
+preemption as the victim. During a preemption process, if the preemptor is configured with the `PreemptionPolicy` and the victim is 
+configured with the `AllowDisruptionByPriorityGreaterThanOrEqual`, the `AllowDisruptionByPriorityGreaterThanOrEqual` takes priority over 
+the `PreemptionPolicy`.
```

Comment:
We may rephrase this as PreemptionPolicy only applies wgeb priority is lower or higher